sympy/testing/tests/test_pytest.py

```diff
@@ -43,7 +43,7 @@ def test_expected_exception_is_silent_with():
 def test_lack_of_exception_triggers_AssertionError_with():
     try:
         with raises(Exception):
-            1 + 1
+            _ = 1 + 1
```

Comment:
This kind of thing is less good but I don't know if there's a way to detect this.

Reply:
I'm not sure yet how to handle visitation of a "with" ( or "if") node. If the node includes the head and body of the block then I need to recurse into the body. Perhaps we can ignore contents of "with" blocks. But we need to look at the body of "if" blocks (per note in doctest).

Reply:
expressions in `with-blocks` are not tested